File: src/main/java/net/arvandor/numinoustreasury/item/action/Blocked.java
package net.arvandor.numinoustreasury.item.action;

import org.bukkit.configuration.serialization.SerializableAs;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.event.player.PlayerItemConsumeEvent;

import java.util.Map;

@SerializableAs("Blocked")
public final class Blocked implements NuminousOnEat, NuminousOnInteractAir, NuminousOnInteractBlock {

    @Override
    public void onEat(PlayerItemConsumeEvent event) {
        event.setCancelled(true);
    }

    @Override
    public void onInteractAir(PlayerInteractEvent event) {
        event.setCancelled(true);
    }

    @Override
    public void onInteractBlock(PlayerInteractEvent event) {
        event.setCancelled(true);
    }

    @Override
    public Map<String, Object> serialize() {
        return Map.of();
    }

    public static Blocked deserialize(Map<String, Object> serialized) {
        return new Blocked();
    }

}

File: src/main/kotlin/net/arvandor/numinoustreasury/item/action/Blocked.kt
package net.arvandor.numinoustreasury.item.action

import org.bukkit.configuration.serialization.SerializableAs
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.event.player.PlayerItemConsumeEvent

@SerializableAs("Blocked")
class Blocked : NuminousOnEat, NuminousOnInteractAir, NuminousOnInteractBlock {
    override fun onEat(event: PlayerItemConsumeEvent) {
        event.isCancelled = true
    }

    override fun onInteractAir(event: PlayerInteractEvent) {
        event.isCancelled = true
    }

    override fun onInteractBlock(event: PlayerInteractEvent) {
        event.isCancelled = true
    }

    override fun serialize(): Map<String, Any> {
        return mapOf()
    }

    companion object {
        @JvmStatic
        fun deserialize(serialized: Map<String?, Any?>?): Blocked {
            return Blocked()
        }
    }
}
